dashboard/app/assets/javascripts/header.js

```diff
@@ -415,35 +412,48 @@ dashboard.header.updateTimestamp = function () {
 };
 
 /**
- * Get the user progress for an anonymous user from the client side cookie.
- * @param {string} scriptName The script to get progress for.
- * @return Object that is a representation of the user's individual level progress.
+ * See ApplicationHelper#activity_css_class.
+ * @param result
+ * @return {string}
  */
-function getSummarizedProgressForAnonymousUser (scriptName) {
-  var summarizedProgress = {};
-  var levelProgress = {};
+function activityCssClass(result) {
+  if (!result) return 'not_tried';
+  if (result >= 1000) return 'submitted';
+  if (result >= 30) return 'perfect';
+  if (result >= 20) return 'passed';
+  return 'attempted';
+}
 
-  summarizedProgress.lines = dashboard.clientState.lines();
+/**
+ * Returns the "best" of the two results, as defined in apps/src/constants.js.
+ * Note that there are negative results that count as an attempt, so we can't
+ * just take the maximum.
+ * @param {Number} a
+ * @param {Number} b
+ * @return {string} The result css class.
+ */
+function mergeProgress(a, b) {
```

Comment:
The name of this should indicated that it returns a css class; e.g. "mergedActivityCssClass"